contracts/Example3.sol

```diff
@@ -1,14 +1,19 @@
+// SPDX-License-Identifier: MIT
 pragma solidity ^0.8.4;
 
 contract Example3 {
-    uint256 marginPercentage = 30;
+    //uint256 marginPercentage = 30;
     function getOwnerMargin(uint256 amount)
         public
-        view
+        pure
         returns (uint256 amountForSender, uint256 amountForOwner)
     {
-        amountForSender = (amount * (100 - marginPercentage)) / 100;
-        amountForOwner = (amount * marginPercentage) / 100;
+        amountForSender = (amount * (100 - 30)) / 100;
```

Comment:
It is not a right practise to hard code the value of `marginPercentage` here. The expected optimisation was to store `marginPercentage` memory since it it being accessed twice from storage

Reply:
Were you able to check that after storing it into memory in example 2, the gas was optimised?

Reply:
For comment on example 3 - Agree that it's not right to use magic numbers in code but when I experimented I see that it saved gas so used it :)

For example 2 - 

    function incrementBy(uint[] calldata arr) external {
        //uint sum = counter;
        //uint len = arr.length;
        for (uint idx = 0; idx < arr.length; ++idx){
            counter += arr[idx];
        }
        //counter = sum;
    }

I tried using local variable sum to store intermediate sum (note that the loop body was sum += arr[i] and the  only one store operation to counter. I expected this to decrease gas but to my surprise gas cost actually went up. Any idea why?

I also used local variable to store arr.length so that we don't need to do it on each iteration but that also increased gas cost. I was a bit lost there